src/compiler/parser.ts

```diff
@@ -2704,6 +2704,14 @@ namespace ts {
                 return arrowExpression;
             }
 
+            // From ES6 spec:
+            // AsyncArrowFunction[In, Yield, Await]::

```

Comment:
how is this presented in the spec? I feel like it should be integrated into the grammar production list (1,2,3,4,5) above, as well as in the code. For example, it could be

```ts
const arrowExpression = tryParseParen... || tryParseUnparen ...;
if (arrowExpression) { return arrowExpression; }
```